frontend/public/components/build-pipeline.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { fromNow } from './utils/datetime';
+import { K8sResourceKind } from '../module/k8s';
+
+const getBuildNumber = (resource: K8sResourceKind): number => _.get(resource, ['metadata', 'annotations', 'openshift.io/build.number']);
+const getStages = (status): any[] => (status && status.stages) || [];
+const getJenkinsStatus = (resource: K8sResourceKind) => {
+  const status = _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-status-json']);
+  if (!status) {
+    return {};
+  }
+
+  const json = _.attempt(JSON.parse, status);
```

Comment:
nit: It's no longer JSON now that you've parsed it... I'd swap these variable names.

```js
const json = _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-status-json']);
// ...
const status = _.attempt(JSON.parse, json);
```

Reply:
Got it.